default_warehouse_from_sale_team/models/default_warehouse_mixin.py

```diff
@@ -0,0 +1,63 @@
+from odoo import api, models
+
+
+class DefaultWarehouseMixin(models.AbstractModel):
+    """If you inherit from this model and add a field called warehouse_id into
+    the model, then the default value for such model will be the one
+    set into the sales team.
+
+    Make sure to put this mixin at the top of inheritance (before the base model), e.g.
+
+        _inherit = ["default.warehouse.mixin", "sale.order"]
+    """
+    _name = "default.warehouse.mixin"
+    _description = "Default Warehouse"
+
+    @api.model
+    def default_get(self, fields_list):
+        """Force that if model has a field called warehouse_id the default
+        value is the one set in the user's sales team.
+        """
+        defaults = super().default_get(fields_list)
+        default_warehouse = self.env.user._get_default_warehouse_id()
+        if not default_warehouse:
+            return defaults
+        warehouse_fields = self.env['ir.model.fields'].sudo().search([
+            ('model', '=', self._name),
+            ('relation', '=', 'stock.warehouse'),
+            ('ttype', '=', 'many2one'),
+            ('name', 'in', list(defaults)),
```

Comment:
```suggestion
            ('name', 'in', fields_list),
```

Otherwise only already-set values will be considered

Reply:
Done!